src/pt/Kairostoons/src/eu/kanade/tachiyomi/extension/pt/Kairostoons/Kairostoons.kt

```diff
@@ -0,0 +1,197 @@
+package eu.kanade.tachiyomi.extension.pt.Kairostoons
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Kairostoons : ParsedHttpSource() {
+
+    override val name = "Kairostoons"
+    override val baseUrl = "https://kairostoons.net"
+    override val lang = "pt"
+    override val supportsLatest = true
+
+    private val dateFormat = SimpleDateFormat("dd/MM/yyyy", Locale("pt", "BR"))
+
+    // =========================================================================
+    //  Popular
+    // =========================================================================
+
+    override fun popularMangaRequest(page: Int): Request {
+        val url = if (page == 1) {
+            "$baseUrl/manga/todos/"
+        } else {
+            "$baseUrl/manga/todos/page/$page/"
+        }
+        return GET(url, headers)
+    }
+
+    override fun popularMangaSelector() = "article.comic-card, div.bsx"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        val anchor = element.selectFirst("a")
+
+        if (anchor != null) {
+            setUrlWithoutDomain(anchor.attr("href"))
+
+            title = element.selectFirst("h3.comic-card-title, .tt")?.text()?.trim()
+                ?: anchor.attr("title")
+        }
```

Comment:
```suggestion
        setUrlWithoutDomain(anchor!!.attr("href"))
        title = element.selectFirst("h3.comic-card-title, .tt")?.text()
                ?: anchor!!.attr("title")
```
url and title is mandatory